include/crocoddyl/core/activations/smooth-abs.hpp

```diff
@@ -30,20 +42,40 @@ class ActivationModelSmoothAbsTpl : public ActivationModelAbstractTpl<_Scalar> {
   typedef typename MathBase::VectorXs VectorXs;
   typedef typename MathBase::MatrixXs MatrixXs;
 
-  explicit ActivationModelSmoothAbsTpl(const std::size_t& nr) : Base(nr){};
+  /**
+   * @brief Initialize the smooth-abs activation model
+   *
+   * The default `eps` value is defined as 1.
+   *
+   * @param[in] nr   Dimension of the residual vector
+   * @param[in] eps  Smoothing factor (default: 1.)
+   */
+  explicit ActivationModelSmoothAbsTpl(const std::size_t& nr, const Scalar& eps = Scalar(1.)) : Base(nr), eps_(eps){};
   virtual ~ActivationModelSmoothAbsTpl(){};
 
+  /**
+   * @brief Compute the smooth-abs function
+   *
+   * @param[in] data  Smooth-abs activation data
+   * @param[in] r     Residual vector \f$\mathbf{r}\in\mathbb{R}^{nr}\f$
+   */
   virtual void calc(const boost::shared_ptr<ActivationDataAbstract>& data, const Eigen::Ref<const VectorXs>& r) {
     if (static_cast<std::size_t>(r.size()) != nr_) {
       throw_pretty("Invalid argument: "
                    << "r has wrong dimension (it should be " + std::to_string(nr_) + ")");
     }
     boost::shared_ptr<Data> d = boost::static_pointer_cast<Data>(data);
 
-    d->a = (r.array().cwiseAbs2().array() + Scalar(1)).array().cwiseSqrt();
+    d->a = (r.array().cwiseAbs2().array() + eps_).array().cwiseSqrt();
```

Comment:
This is not the implementation that we discussed.
What we need is sqrt(eps+r**2). This would be somewhat like this:
```
d->a = sqrt(r.array().cwiseAbs2().sum() + 1);
```

Please update the doc for this function as well

Reply:
Carlos, the current implementation is wrong.
1) It returns an array, and assigns it to a scalar.
2) It doesn't do the computation needed.

What is expected from this function is sqrt(eps+r**2). I think we should change to that.

Reply:
Think about it in another way.
r is a scalar value, and then you apply the smooth-abs function for each single coefficient of the residual vector.

This is not wrong neither right. There are only two different approaches.

Btw, this is the original implementation of the smooth-abs function. I didn't changed it when I was coding the c++ version.

Reply:
Well looking twice, I see it was changed but frankly speaking I don't remember I did.
See https://github.com/loco-3d/crocoddyl/blob/master/unittest/python/crocoddyl/activation.py#L160

Reply:
After internal discussion the formulation should be kept as implemented.